subsys/net/lib/lwm2m_client_utils/Kconfig

```diff
@@ -7,6 +7,10 @@
 menuconfig LWM2M_CLIENT_UTILS
 	bool "LWM2M client utilities library"
 	select LWM2M if !ZTEST
+	imply LTE_LC_EDRX_MODULE
+	imply LTE_LC_NEIGHBOR_CELL_MEAS_MODULE
+	imply LTE_LC_PSM_MODULE
+	imply LTE_LC_TAU_PRE_WARNING_MODULE
```

Comment:
as raised else-where in comments, then I'm a bit unsure if lwm2m actually depends on lte lc modules or not.

And if lwm2m works just as fine with lte lc modules as without, then why the `imply` ?
I do notice that changing this to a `depends on` seems to open pandoras box of circular dependencies (including select and imply deps), so far seem to be originating at a bad `select FLASH` somewhere (still investigating).

But I would like some more reasoning regarding those `imply`, than just, `this makes it work`.

Reply:
@tejlmand `lwm2m_client_utils` depends on the link controller, but it doesn't show. This is happening because, as far as I understand, the way it is built indirectly brings in the link controller. This is a design mistake because each component should export a contract of its dependencies, at minimum to document them.

Reply:
accepted. To be addressed later.
Please create internal ticket.